core/src/main/scala/spark/storage/BlockManager.scala

```
@@ -382,6 +383,8 @@ private[spark] class BlockManager(
     // As an optimization for map output fetches, if the block is for a shuffle, return it
     // without acquiring a lock; the disk store never deletes (recent) items so this should work
     if (ShuffleBlockManager.isShuffle(blockId)) {
+      //close the shuffle Writers for blockId
```

Do shuflfe blocks still walk through this code path? If no, I would just throw an exception if it is a shuffle block here.